packages/sdk/contractkit/src/kit.ts

```diff
@@ -222,6 +223,74 @@ export class ContractKit {
     return blockchainParamsWrapper.getEpochNumberOfBlock(blockNumber)
   }
 
+  /*
+   * Sets the maxFeeInFeeCurrency on the provided transaction
+   *
+   * @remarks
+   * Because tx with maxFeeInFeeCurrency set require maxPriorityFeePerGas and maxFeePerGas to be expressed in CELO
+   * and previous tx types with feeCurrency expressed these in the token paying for the transaction,
+   * this method overwrites maxPriorityFeePerGas and maxFeePerGas to ensure they are correctly valued in CELO
+   *
+   * @param tx.gas is required
+   */
+  async populateMaxFeeInToken(tx: CeloTx & Pick<Required<CeloTx>, 'feeCurrency'>): Promise<CeloTx> {
+    if (!(await isCel2(this.connection.web3))) {
+      throw new Error("Can't populate `maxFeeInFeeCurrency` if not on a CEL2 network")
+    }
+
+    if (isPresent(tx.feeCurrency) && !isPresent(tx.maxFeeInFeeCurrency)) {
+      if (!isPresent(tx.gas)) {
+        tx.gas = await this.connection.estimateGas(tx)
+      }
+
+      // Force maxFeePerGas and maxPriorityFeePerGas to be expressed in CELO
+      const [maxFeePerGas, maxPriorityFeePerGas] = await Promise.all([
+        this.connection.gasPrice(),
+        await this.connection.rpcCaller.call('eth_maxPriorityFeePerGas', []).then((x) => x.result),
+      ])
+      tx.maxFeePerGas = new BigNumber(maxFeePerGas).toString(10)
+      tx.maxPriorityFeePerGas = new BigNumber(maxPriorityFeePerGas).toString(10)
+      const maxFeeInFeeCurrency = await this.estimateMaxFeeInFeeToken({
+        feeCurrency: tx.feeCurrency,
+        gasLimit: BigInt(tx.gas),
+        maxFeePerGas: BigInt(maxFeePerGas),
+      })
+
+      tx.maxFeeInFeeCurrency = maxFeeInFeeCurrency.toString(10)
+    }
+
+    return tx
+  }
+
+  /**
+   * For cip 66 transactions (the prefered way to pay for gas with fee tokens on Cel2) it is necessary
+   * to provide the absolute limit one is willing to pay denominated in the token.
+   * In contrast with earlier tx types for fee currencies (celo legacy, cip42, cip 64).
+   *
+   * Calulating Estimation requires the gas, maxfeePerGas and the conversion rate from CELO to feeToken
+   * https://github.com/celo-org/celo-proposals/blob/master/CIPs/cip-0066.md
+   */
+  async estimateMaxFeeInFeeToken({
```

Comment:
could we maybe add more test cases for different exchange rates than the one provided by the directory right now? I feel like it'd be good to have coverage for different cases than just 1:1 ratio